### ISSUE TITLE
[CAIA a11y Research - PILOT]: Disability Benefits Experience Team 1 (DBEX 1), 526ez (Mobile) (Nov. 2024)

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#97151]
> [!IMPORTANT]
> - **TICKET STATUS:** `Completed`
> - **PILOT DATE:**  11/21/2024 from 9-11 a.m. EST 
> - **DEVICE:** Mobile
> - **TYPES OF ASSISTIVE TECH:** iPhone + VoiceOver
> - **ENVIRONMENT:** Staging

# CAIA A11y Support
This is a child ticket of the research ticket [96810](https://github.com/department-of-veterans-affairs/va.gov-team/issues/96810) - created to track work, completed from this pilot session.
> [!TIP]
> **DEFINITION OF DONE**: Refer to the task list in this ticket, for actionable items.

## CAIA Assignee(s)

<details><summary>Toggle to view CAIA info</summary>

- @mutiaadhocteam (pilot)
- @sara-amanda (co-pilot)

</details>

## Team Info

<details><summary>Toggle to view team info</summary>

- Team name: Disability Benefits Experience Team 1
- OCTO-DE product owner: Emily Theis
- Product manager: Jared Pace
- Researchers: Maya Carroll, Evelyn Hilbrich Davis, Lauren Dawson
- FE engineer: Christine Cereca, Tracy Tran
- Product/team Slack channel: benefits-disability

</details>

## Description

<details><summary>Toggle to view the description</summary>

**A CAIA a11y will participate as a user in a pilot of this research study.**

- **Researcher**: The VFS team researcher will test their prototype and conversation guide by conducting the pilot session with a CAIA A11Y.
- **User**: The CAIA A11Y team member will act as a real user in the pilot session.
- **Assistive Tech**: The CAIA A11Y team member will follow the prompts given by the VFS Researcher, while using and demonstrating how the behavior of a `screen reader` would behave.

### Pilot Session
- **Date:** 11/21/2024
- **Prototype**: [Mobile Prototype Link](https://staging.va.gov/disability/file-disability-claim-form-21-526ez/introduction )
- **Location:** Staging
- **Available after 11/18/2024**
    - [Research Plan](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/research/2024-11-Toxic%20Exposure/AT-%20Research%20Plan.md)
    - [Conversation Guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/research/2024-11-Toxic%20Exposure/AT-%20Conversation%20Guide.md)
</details>

### Assistive Tech Options

> [!TIP]
> **DEVICE SUPPORT DURING RESEARCH SESSIONS**
> - **Available AT**: `Screen readers` on a `desktop` device is currently the primary source of AT that CAIA A11ys are able to support, in addition to `screen magnification` for user research sessions.
> - **Additional Specifications:** If there is a particular `test user`, `browser` or `operating system` you would like us to use, please let us know in advance via this ticket's comments, tagging the assignee - or reach out to @coforma-terry in the CAIA Slack Channel: [#sitewide-content-accessibility-ia ](https://dsva.slack.com/archives/C01K37HRUAH)

## Definition of Done

```[tasklist]
### A11y Tasks
- [x] **Schedule CAIA a11y:**
- [x] **Share link to [Assistive Tech Pilot Guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/CAIA/accessibility/assistive-tech-pilot-guide.md) with team, if they haven't reviewed it yet**
- [x] **Determine date and time for pilot:**
- [x] **Update CAIA A11y Capacity Calendar:** Add invite for pilot, when scheduled and dcvd
- [x] **Add to research tracker**
- [x] **Review research** [conversation guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/research/2024-11-Toxic%20Exposure/AT-%20Conversation%20Guide.md) and provide feedback in comments.
- [x] [Request AT device and tech to be used in sessions](LINK)
- [x] [Receive AT device and tech from Team in research ticket](LINK) to be used in sessions to prep for pilot
- [x] **Review prototype** in [Staging](https://staging.va.gov/disability/file-disability-claim-form-21-526ez/introduction) (users 54 or 229)
- [x] **Run pilot** with VFS Researcher
- [x] **Optional:** [Note Taking](https://docs.google.com/document/d/1QFs5mDiVO1kgoOi4GpOgnDvkktznmnjCBDn8dza016Q/edit?tab=t.0#heading=h.7b53y4dqihzk)
```

**Comment by sara-amanda** [comment]
## Scheduling Pilot(s)

Evelyn Hilbrich Davis

> Hello, I would like to schedule our 2 pilot sessions for mobile/desktop. We are planning the sessions for 90 minutes.
> Friday 11/22, 11:30-3pm EST
> Monday 11/25 1-3pm EST
> Tuesday 11/26 1-3pm EST.
> [@sitewide-caia-accessibility](https://dsva.slack.com/admin/user_groups) :thank_you:

cc: @coforma-jamie

**Comment by sara-amanda** [comment]
# Pilot Takeaways

- **Type**: Mobile Pilot
- **Device**: iPhone (iOS)
- **Browser**: Safari
- **AT**: VoiceOver
- **Prototype Environment**: Staging

## Findings

### Error Handling
#### Dates & Locations
- **Findings:** 
    - The error handling for the dates in accordance with the locations was confusing. 
    - As it currently stands, the announcements of errors while the user is attempting to fill out the date is disruptive to the experience.
- **Recommendation:** 
    - The error handling should be tied to the continue button, not the date picker or fields for the dates themselves. 

#### Timeframes
- **Findings:** 
    - It is unclear whether the timeframes are necessary to add to the form. No error is announced when the dates are left empty.
    - It is also confusing that errors are only triggered when a date is left incomplete.
- **Recommendations:**
    - The error should only be announced if it is required. Also if it is unnecessary. Then it should say that it is unnecessary. Because even without star req, some users may believe it is necessary to add time frame, because usually timeframe is necessary. 
    - In regard to errors being triggered if date is incomplete, the error trigger should be more specific (please specify a date).
    - **Scenario**: Some users may start typing in dates, but then realize that they don’t know it fully and instead of removing the content  they may check the box that they don’t know. The current behavior is you can’t fill it out partially and forget the dates. 
- **Next Steps:** _The team is working on this for next version. Removing the month and day._

#### Maximum Characters
- **Finding:** 
    - I would think it would give me an error if there was over the 250 maximum? How do you let the user with AT know they have only a few characters left? 
- **Recommendations:** 
    - Possibility, validating how much you have left to type below the box that can be announced when focus is moved from the typing box, as to not interrupt.     
    - - **Example**: 230/250, allowing the user to go back and type more. 
        - DO NOT announce it while typing, only when navigating to it below the box. 
    - **_It can help to manage expectations of text remaining._** 



### Required Field
####  Star Required
- **Finding**:
    - Required is not connected - it is separate.
- **Recommendations:** 
    - **In the Code**: Generally, if any field or set of data is required, then “*required” should be announced and tied to the group to inform the user that the information is required.
    - **Optional?** AT users would like to know if it is not required to manage expected behavior. 
        - _**Optional** **or** **Required** to reduce confusion around future errors._ 

### Navigating & Placement 
#### Form Summary
- **Finding**: 
    - The path to return to the form summary before submission is unclear. 
    - Users will find it cumbersome to navigate through an extraneous number of pages just to return to the page.
- **Recommendation:** 
    - There should be a shortcut button to return to the form summary / final page before submission. 

#### Notifications
- **Finding**: 
    - The alert for autosaving is set at the bottom and after the continue button. 
    - While this may be clear for sighted users, blind and visually impaired users will have difficulty recognizing this alert due to its position on the screen.
- **Recommendation:** 
    - The autosaving notification should be above the continue button to avoid being missed, entirely. 

#### Back Button
- **Findings**:  
    - The back button is located near the bottom of the form. 
    - Users will be required to navigate through all fields, texts, components, etc. just to return to the previous screen. 
    - While native gestures exist to return to the previous screen, not all users are proficient enough to use these native gestures.
- **Recommendations:** 
    - The back button should be located near the top of the form. 
    - **Scenario:** _You can click continue through the form until you get back to the review page._
        - A shortcut could exist to take you back to the conditions page, to save time and clicking. 

#### Progress Bar
- **Findings**: 
    -  Staying in step 2 of 5 for multiple screens can be confusing to users. 
- **Recommendation:** 
    - Could there be a progress bar that shows you where you are within step 2 of 5 since the step is not changing, despite the screens? 
        - Or, can you do step 2 of 5 and then have section A, B or C? 
        - Or a label for the sub-sections to show progress? 
            - Section 1 of 3 within Step 2 of 5? 

> “I hate that it says 2 of 5 conditions and it feels like I haven’t moved an inch. It is disorienting” - Maria



## Positive Feedback

> The specifications for the functions of the fields in the accessible names (e.g. “Edit date for X”) was great! - Maria

### Additional Positive Feedback
> - Expand all button for the accordions, much easier than expanding 1 by 1. 
> - Love how they use expandable components attached to heading levels, it tells the user the expected structure of the content and what you can use them for, without being too overwhelming to the user.
> - Privacy policy opens in the new tab (Maria likes that)
> - Edit vs Edit Payment information is very helpful.
> - Information is very streamlined.
> - The fact that it said new, vs answer our questions, it gives a sense of urgency and necessity.
> - Navigating to new questions: 5 Very Easy. Placement was logical and it made sense for where it was set in the accordions.
> - I like how it says if things are checked or not checked.
> - Alpha order helps - I kind of know what to expect next.

 
cc: @mutiaadhocteam

**Comment by evelynhilbrichdavis** [comment]
@pacerwow Here is the CAIA ticket I recommend we attach to the Post-MVP Enhancements. This notes came from a CAIA pilot research session, but also offer great insight into how to improve the a11y experience of the user flow.
